src/main/kotlin/ai/digital/integration/server/deploy/internals/cluster/operator/OperatorHelper.kt

```diff
@@ -40,47 +47,117 @@ abstract class OperatorHelper(val project: Project) {
+            "pod/dai-xld-digitalai-deploy-master-$position"
+        }
+
+        resources.forEach { resource ->
+            if (!KubeCtlUtil.wait(project, resource, "Ready", getProfile().deploymentTimeoutSeconds.get())) {
+                throw RuntimeException("Resource $resource is not ready")
+            }
+        }
+    }
+
+    fun waitForWorkerPods() {
+        val resources = List(getWorkerCount()) { position ->
+            "pod/dai-xld-digitalai-deploy-worker-$position"
+        }
+        resources.forEach { resource ->
+            if (!KubeCtlUtil.wait(project, resource, "Ready", getProfile().deploymentTimeoutSeconds.get())) {
+                throw RuntimeException("Resource $resource is not ready")
+            }
+        }
+    }
+
+    fun waitForBoot(host: String) {
+        val url ="http://$host/xl-deploy/deployit/metadata/type"
+        val server = DeployServerUtil.getServer(project)
+        WaitForBootUtil.byPort(project, "Deploy", url, null, server.pingRetrySleepTime, server.pingTotalTries)
+    }
+
+    fun undeployCis() {
+        val fileStream = {}::class.java.classLoader.getResourceAsStream("operator/python/undeploy.py")
+        val resultComposeFilePath = Paths.get(getProviderWorkDir(), "undeploy.py")
+        fileStream?.let {
+            FileUtil.copyFile(it, resultComposeFilePath)
+        }
+        try {
+            CliUtil.executeScripts(project, listOf(resultComposeFilePath.toFile()), "undeploy.py", false, 4516)
+        } catch (e: RuntimeException) {
+            project.logger.warn("Undeploy didn't run. Check if operator's deploy server is running on port 4516: ${e.message}")
+        }
+    }
+
+    open fun getOperatorImage(): String {
+        return getProvider().operatorImage.value("xebialabs/deploy-operator:1.2.0").get()
+    }
+
     fun updateOperatorCrValues() {
         val file = File(getProviderHomeDir(), OPERATOR_CR_VALUES_REL_PATH)
         val pairs = mutableMapOf<String, Any>(
-            "spec.ImageTag" to DeployServerUtil.getServer(project).version!!,
-            "spec.KeystorePassphrase" to getProvider().keystorePassphrase,
-            "spec.Persistence.StorageClass" to getStorageClass(),
-            "spec.RepositoryKeystore" to getProvider().repositoryKeystore,
-            "spec.postgresql.persistence.storageClass" to getStorageClass(),
-            "spec.rabbitmq.persistence.storageClass" to getStorageClass(),
-            "spec.rabbitmq.persistence.replicaCount" to "1",
-            "spec.route.hosts" to arrayOf(getProvider().host),
-            "spec.xldLicense" to getLicense()
+                "spec.ImageRepository" to DeployServerUtil.getServer(project).dockerImage!!,
+                "spec.ImageTag" to DeployServerUtil.getServer(project).version!!,
+                "spec.XldMasterCount" to getMasterCount(),
+                "spec.XldWorkerCount" to getWorkerCount(),
+                "spec.Persistence.XldMasterPvcSize" to "10Gi",
```

Comment:
Why so much memory? We only running tests there :) 
Can we use like 100Mi or less?

Reply:
This is actually the original operator value, I was playing with that yesterday just to see if the operator is faster in a startup with different values.